{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "provenance": []
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    }
  },
  "cells": [
    {
      "cell_type": "markdown",
      "source": [
        "## Task 1.1 Summarize the GWAS results\n"
      ],
      "metadata": {
        "id": "-ShR37Z8reyj"
      }
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "9-WtMrHSsHiu"
      },
      "source": [
        "In this first part of the assignement, we are going to load the summary results from a Genome-Wide Association Study (GWAS) and examine its results. In the second part, we examine the mouse homologous gene."
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "### THE TEXT AND CODE IN THIS FIELD IS PROVIDED TO THE STUDENTS\n",
        "#\n",
        "### Read ALL instructions FIRST before you begin your data analysis\n",
        "#\n",
        "# We are going to examine the GWAS ouput (i.e. summary statistics) of the 2018 GWAS of Major Depression conducted by the PGC.\n",
        "# The data belong to this publication: Nat Genet. 2018 May;50(5):668-681. doi: 10.1038/s41588-018-0090-3. PMID:29700475\n",
        "#\n",
        "# Download this summary statistics file:\n",
        "! wget \"https://www.dropbox.com/scl/fi/yj3m8d2puymhdzufanhs9/MDD2018_ex23andMe.txt?rlkey=4dl6gdi977dyj6neuwjtdf1xv&st=82vc8u0o&dl=0\"  -O MDD_2018_original.tsv\n",
        "\n",
        "# Download the README file:\n",
        "! wget \"https://www.dropbox.com/scl/fi/0buvztu2sl6xa12vwvwyb/pgc-mdd-2018-readme.pdf?rlkey=r6qdropq18mi5rowplhoagcp8&st=v4i71xip&dl=0\" -O MDD_2018_README.pdf\n",
        "\n",
        "# Read the README file before moving to Task 1.1\n",
        "\n"
      ],
      "metadata": {
        "id": "YcEHCjJU0yB5",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "6e41a847-8357-4218-ee8e-9de8ae94001b"
      },
      "execution_count": 235,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2025-12-11 16:28:04--  https://www.dropbox.com/scl/fi/yj3m8d2puymhdzufanhs9/MDD2018_ex23andMe.txt?rlkey=4dl6gdi977dyj6neuwjtdf1xv&st=82vc8u0o&dl=0\n",
            "Resolving www.dropbox.com (www.dropbox.com)... 162.125.3.18, 2620:100:601b:18::a27d:812\n",
            "Connecting to www.dropbox.com (www.dropbox.com)|162.125.3.18|:443... connected.\n",
            "HTTP request sent, awaiting response... 302 Found\n",
            "Location: https://uc1597a280f2a69cb34a813f8031.dl.dropboxusercontent.com/cd/0/inline/C21Cn8geOvHU4IjTVn8DA2BjLZLR0E5vY-SeIVm_PGZBvsMrvHPM9sH6G9GWNC1rNsWKUX2NgccLC0i45ykhEfoABTCBOcYVEkpvzR4Gp2m59kFk3kBYfWLoTC28oOqhRbY/file# [following]\n",
            "--2025-12-11 16:28:04--  https://uc1597a280f2a69cb34a813f8031.dl.dropboxusercontent.com/cd/0/inline/C21Cn8geOvHU4IjTVn8DA2BjLZLR0E5vY-SeIVm_PGZBvsMrvHPM9sH6G9GWNC1rNsWKUX2NgccLC0i45ykhEfoABTCBOcYVEkpvzR4Gp2m59kFk3kBYfWLoTC28oOqhRbY/file\n",
            "Resolving uc1597a280f2a69cb34a813f8031.dl.dropboxusercontent.com (uc1597a280f2a69cb34a813f8031.dl.dropboxusercontent.com)... 162.125.3.15, 2620:100:6023:15::a27d:430f\n",
            "Connecting to uc1597a280f2a69cb34a813f8031.dl.dropboxusercontent.com (uc1597a280f2a69cb34a813f8031.dl.dropboxusercontent.com)|162.125.3.15|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 1498171464 (1.4G) [text/plain]\n",
            "Saving to: ‘MDD_2018_original.tsv’\n",
            "\n",
            "MDD_2018_original.t 100%[===================>]   1.39G  17.1MB/s    in 47s     \n",
            "\n",
            "2025-12-11 16:28:52 (30.5 MB/s) - ‘MDD_2018_original.tsv’ saved [1498171464/1498171464]\n",
            "\n",
            "--2025-12-11 16:28:52--  https://www.dropbox.com/scl/fi/0buvztu2sl6xa12vwvwyb/pgc-mdd-2018-readme.pdf?rlkey=r6qdropq18mi5rowplhoagcp8&st=v4i71xip&dl=0\n",
            "Resolving www.dropbox.com (www.dropbox.com)... 162.125.3.18, 2620:100:6023:18::a27d:4312\n",
            "Connecting to www.dropbox.com (www.dropbox.com)|162.125.3.18|:443... connected.\n",
            "HTTP request sent, awaiting response... 302 Found\n",
            "Location: https://uc1c016db02e2d64b043995860ee.dl.dropboxusercontent.com/cd/0/inline/C216ZgtWEB4iUWZFLpeaz3Y9-WrGm6kitCqEc7JysjvZ9_pnM1b9-L4Jk1B820LoqlJOCoLc6PkwPWvIz4USrThyX_4ixeeqd47kqSXHKRnRaMU-Vr0yItGDa6MSsKhhcyM/file# [following]\n",
            "--2025-12-11 16:28:52--  https://uc1c016db02e2d64b043995860ee.dl.dropboxusercontent.com/cd/0/inline/C216ZgtWEB4iUWZFLpeaz3Y9-WrGm6kitCqEc7JysjvZ9_pnM1b9-L4Jk1B820LoqlJOCoLc6PkwPWvIz4USrThyX_4ixeeqd47kqSXHKRnRaMU-Vr0yItGDa6MSsKhhcyM/file\n",
            "Resolving uc1c016db02e2d64b043995860ee.dl.dropboxusercontent.com (uc1c016db02e2d64b043995860ee.dl.dropboxusercontent.com)... 162.125.3.15, 2620:100:6023:15::a27d:430f\n",
            "Connecting to uc1c016db02e2d64b043995860ee.dl.dropboxusercontent.com (uc1c016db02e2d64b043995860ee.dl.dropboxusercontent.com)|162.125.3.15|:443... connected.\n",
            "HTTP request sent, awaiting response... 302 Found\n",
            "Location: /cd/0/inline2/C217Wd_v4_r0TN-QhhizpBfQ_EuVzUzdrFj_zEc-SYvyrJXZNAlCDW17dAIpDVWOY2nFVDcnovg5eJ7zbRh8NpvxnxUMM2iq6wWdMEt5B2vXX8HDk8XJgqh3eck6EXi6gwkI2frFZjRntCEc6lmV-BGaVhwjvcpqjPZYQRqNxjoFcAm2KytY22KN0-3NTO3TUCudsRTeJsk9RmkVT7axt3yLiAMTCHmO_vncgXseGOVVieFBh7IvMQ_7zB7VnNF1fpAWG8iV9b6Lqpnxtb70pHn8jz9AT26MfqMVW9M6XwU2nsA7MtrhrcIExAbvniJRX0VTNoW6sC7U4kRVwqaXpOz8ECAlevbFxa5EnhhtQ3y6Hw/file [following]\n",
            "--2025-12-11 16:28:53--  https://uc1c016db02e2d64b043995860ee.dl.dropboxusercontent.com/cd/0/inline2/C217Wd_v4_r0TN-QhhizpBfQ_EuVzUzdrFj_zEc-SYvyrJXZNAlCDW17dAIpDVWOY2nFVDcnovg5eJ7zbRh8NpvxnxUMM2iq6wWdMEt5B2vXX8HDk8XJgqh3eck6EXi6gwkI2frFZjRntCEc6lmV-BGaVhwjvcpqjPZYQRqNxjoFcAm2KytY22KN0-3NTO3TUCudsRTeJsk9RmkVT7axt3yLiAMTCHmO_vncgXseGOVVieFBh7IvMQ_7zB7VnNF1fpAWG8iV9b6Lqpnxtb70pHn8jz9AT26MfqMVW9M6XwU2nsA7MtrhrcIExAbvniJRX0VTNoW6sC7U4kRVwqaXpOz8ECAlevbFxa5EnhhtQ3y6Hw/file\n",
            "Reusing existing connection to uc1c016db02e2d64b043995860ee.dl.dropboxusercontent.com:443.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 72059 (70K) [application/pdf]\n",
            "Saving to: ‘MDD_2018_README.pdf’\n",
            "\n",
            "MDD_2018_README.pdf 100%[===================>]  70.37K  --.-KB/s    in 0.02s   \n",
            "\n",
            "2025-12-11 16:28:53 (3.39 MB/s) - ‘MDD_2018_README.pdf’ saved [72059/72059]\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "vkXDxly4sM5Y"
      },
      "source": [
        "**Q1.1.** Load the **tab** separated file 'MDD_2025_gwas_data.tsv' into a table and **check the content and the format**. Show your code and explain what you are doing and why. [2 points]\n",
        "\n",
        "Relevant column headers:\n",
        " 1 CHROM: Chromosome;\n",
        " 2\tPOS: Base-pair position (GRCh37);\n",
        " 3\tSNP: SNP Marker ID (rs identifier);\n",
        " 8\tPVAL: P-value;\n",
        "\n",
        "**Q1.2.** How many SNPs were tested?  [1 point]\n",
        "\n",
        "**Q1.3.** Find the **most significant SNP**. What is this top SNP's rs identifier? [1 point]\n",
        "\n",
        "**Q1.4.** Using the NCBI online database, in which gene is SNP rs9479138 located? [0.5 point]\n",
        "\n",
        "**Q1.5.** Looking further in the NCBI database, in which body tissue or organ is this gene most expressed? Provide a link/url showing how you got to this answer. [0.5 point]"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "import pandas as pd\n",
        "df = pd.read_csv('MDD_2018_original.tsv', sep='\\t')"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "0RJC5N31g9yK",
        "outputId": "df4550cb-9440-45d1-cd63-88551b3ebd3e"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "/tmp/ipython-input-1174509403.py:2: DtypeWarning: Columns (0) have mixed types. Specify dtype option on import or set low_memory=False.\n",
            "  df = pd.read_csv('MDD_2018_original.tsv', sep='\\t')\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "df.info()\n",
        "print(df.head())\n",
        "df.shape"
      ],
      "metadata": {
        "id": "CPs5jnFEjBFg"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "Q1.1 I loaded the file with pandas and used the function df = pd.read_csv('MDD_2018_original.tsv', sep='\\t'). With sep='\\t') I generate seperately tabs for the columns instead with commas. With df.info() and df.head() I get the first 5 rows and overall information about the columns."
      ],
      "metadata": {
        "id": "Ia-nyv7ijQXb"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "print(f\"Total SNPs tested: {df['SNP'].nunique()}\")"
      ],
      "metadata": {
        "id": "RgVZws-rkAHP"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "Q1.2 Total number of unique SNPs tested is 13554489"
      ],
      "metadata": {
        "id": "zf12ArV0l8MI"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "snp = df.loc[df['P'].idxmin()]\n",
        "\n",
        "most_significant_SNP = snp['SNP']\n",
        "most_significant_SNP_p_value = snp['P']\n",
        "\n",
        "print(f\"Most significant SNP: {most_significant_SNP}\")\n",
        "print(f\"p-value: {most_significant_SNP_p_value}\")\n",
        "\n"
      ],
      "metadata": {
        "id": "nwDfVGKJmEkw"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "Q1.3 The most signficant SNP is rs76025409 with a p-value of 2.135e-11.\n",
        "\n",
        "Q1.4 SNP rs9479138 is located in the ESR1 (Estrogen Receptor 1) gene region at Chr6\n",
        "\n",
        "Q1.5 The ESR1 gene is most expressed based on the RNA of normal human tissues in the uterus and endometrium (HPA) https://www.ncbi.nlm.nih.gov/gene/2099#gene-expression\n"
      ],
      "metadata": {
        "id": "ynNYjZBpoZH5"
      }
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "3ePU355NsZ7d"
      },
      "source": [
        "## Task 1.2 Microarray database\n",
        "Another gene that has often been associated with Major Depression is **PGBD1**.In the Allen Institute microarray database, search for expression patterns of PGBD1: https://human.brain-map.org/microarray/search.\n",
        "\n",
        "Download the relevant data for PGB1 (first probe, first row). There are two ways to do this:\n",
        "1. Select the top row (probe) and download this data. Upload the data into  Colab by clicking on the 'file' icon on the top left corner.\n",
        "2. Run the code provided below using wget commands (pointing to a dropbox database)."
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#### This is provided to the students! #######\n",
        "#downloading data\n",
        "! wget 'https://www.dropbox.com/scl/fi/th8ykvjnk9rlao0ccp47d/Columns.csv?rlkey=dttm23lamkhugkypb0r5bgaj2&st=1wkbgc7r&dl=0' -O Columns.csv\n",
        "! wget 'https://www.dropbox.com/scl/fi/xs9g0r3egrxi93hhn9pel/Expression.csv?rlkey=jf47vz3ez0mxeascgjdqjn6rc&st=92e3dfew&dl=0' -O Expression.csv\n",
        "! wget 'https://www.dropbox.com/scl/fi/j2cn98z1vu5igppx69ayf/Contents.txt?rlkey=ew526p39wpe78os3x46rzglrg&st=wd6jzr40&dl=0' -O Contents.txt"
      ],
      "metadata": {
        "id": "DspeMjifbHbp"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "**Read the \"Contents.txt\" file.**\n",
        "\n",
        "**Load the Expression.csv file and the Columns.csv file in python.**\n",
        "\n",
        "**Q1.6**  Format the data of Expression.csv and Columns.csv so that you have a table with 1 column of PGB1 expression values, next to columns with information of regions and subjects (donors). You should have 1 row per brain structure for each subject. (1 point)\n",
        "\n",
        "**Q1.7** Calculate the mean expression accross participants per brain structure.  In which brain structure does this PGB1 have the highest mean expression? (0.5 point)\n",
        "\n",
        "**Q1.8**   How many missing values are there for the expression of this gene across all brain regions? And for the brain region with the highest expression? (0.5 point)\n",
        "\n",
        "**Q1.9**  Does PGBD1 look consistently highly expressed across subjects in this top brain region? For full points, consider patterns of PGBD1 expression across different subjects and brain regions. Use figures and/or statistics to help you decide. (2 points)\n",
        "\n",
        "**Q1.10**   Looking at a coarser level of analysis, in which (larger) brain structures is PGBD1 most highly expressed? (1 point)"
      ],
      "metadata": {
        "id": "T-2hdvEEryGO"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "df_col = pd.read_csv('Columns.csv')\n",
        "df_exp = pd.read_csv('Expression.csv')\n",
        "\n",
        "df_col.shape\n",
        "df_exp.shape"
      ],
      "metadata": {
        "id": "AAgFLMr9rAMv"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "import pandas as pd\n",
        "import numpy as np\n",
        "\n",
        "df_exp_transposed = df_exp.T\n",
        "df_exp_transposed = df_exp_transposed.reset_index()\n",
        "df_exp_transposed.columns = [\"PGBD1\"]\n",
        "df_exp_transposed = df_exp_transposed.iloc[1:].reset_index(drop=True)\n",
        "\n",
        "print(df_exp_transposed)\n",
        "print(df_exp_transposed.shape)"
      ],
      "metadata": {
        "id": "ht-5AVCI6k4q"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "df_col[\"PGBD1\"] = df_exp_transposed[\"PGBD1\"]\n",
        "df_col.head(10)"
      ],
      "metadata": {
        "id": "2CY35hPgOYD4"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "df_col['PGBD1'] = pd.to_numeric(df_col['PGBD1'], errors='coerce')\n",
        "print(df_col['PGBD1'].dtype)"
      ],
      "metadata": {
        "id": "6qBX2ymqKqaf"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "mean_expr = df_col.groupby(\"structure_name\")[\"PGBD1\"].mean().sort_values(ascending=False)\n",
        "print(mean_expr)\n"
      ],
      "metadata": {
        "id": "Up3TpLePJnqW"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "Q1.7 Highest mean expression is in the VIIB region."
      ],
      "metadata": {
        "id": "MmM3CO96R3rR"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "missing_values = df_col['PGBD1'].isnull().sum()\n",
        "print(f\"Number of missing values: {missing_values}\")\n",
        "\n",
        "missing_values_VVIB = df_col[df_col['structure_name'] == 'VIIB']['PGBD1'].isnull().sum()\n",
        "print(f\"Number of missing values in VVII region: {missing_values_VVIB}\")"
      ],
      "metadata": {
        "id": "yOC2BAfOSKEX"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "Q1.8 The total of missing values for all the regions is 181, for the region with the highest expression (VIIB) 3 values are missing"
      ],
      "metadata": {
        "id": "c6F8t-GeSwIA"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "import matplotlib.pyplot as plt\n",
        "import seaborn as sns\n",
        "\n",
        "PGBD1_expression = df_col[df_col['top_level_structure_name'] == 'cerebellar cortex']['PGBD1']\n",
        "\n",
        "subregions = ['I', 'II', 'III', 'IV', 'V', 'VI', 'VIIAf', 'VIIAt', 'VIIB', 'VIIIA', 'VIIIB', 'IX', 'X', 'Crus I', 'Crus II']\n",
        "cerebellar_subregions_df = df_col[df_col['structure_name'].isin(subregions)]\n",
        "subregion_stats = cerebellar_subregions_df.groupby('structure_name')['PGBD1'].describe()[['count', 'mean', 'std', 'min', 'max']]\n",
        "subregion_stats_sorted = subregion_stats.sort_values(by='mean', ascending=False)\n",
        "\n",
        "print(subregion_stats_sorted)\n",
        "\n",
        "\n",
        "sns.boxplot(x='structure_name', y='PGBD1', data=cerebellar_subregions_df, showcaps=True, showfliers=False)\n",
        "\n",
        "plt.xticks(rotation=90)\n",
        "plt.ylabel('PGBD1 Expression (z-value)')\n",
        "plt.xlabel('Cerebellar Cortex Subregions')\n",
        "plt.title('PGBD1 Expression Across Cerebellar Cortex Subregions')\n",
        "plt.tight_layout()\n",
        "plt.show()"
      ],
      "metadata": {
        "id": "p6QqU6_kTZqq"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "Q1.9 It seems to look consistenly highly expressed in this top brain region. In every brain region, the z-value is positive and lays above 0.5."
      ],
      "metadata": {
        "id": "TV1Xa9-ovFEX"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "region_stats = df_col.groupby('top_level_structure_name')['PGBD1'].describe()\n",
        "region_stats_sorted = region_stats.sort_values(by='mean', ascending=False)\n",
        "print(region_stats_sorted[['mean', 'std', 'min', 'max', 'count']])"
      ],
      "metadata": {
        "id": "rLpSj-AarnEa"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "Q1.10 The cerebellar cortex is the average highest expresssed PGBD1 (z-value) brain region with 0.976863 and std of 0.854904 compared to the other top brain regions. However, the cerebellar cortex contains an outlier with a z-value of 7.2349 which makes the average expression higher. Altough the other data points are relatively more close to the average (n = 105). Next follows the cingulate gyrus and insula. In contrast to these regions, the cerebellar cortex contains 105 values which is way more than the other regions which contain 21 and 11 respectively. I think this is not a fair comparisment for the PGBD1 expression patterns."
      ],
      "metadata": {
        "id": "KOsB6v9lorKO"
      }
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "jzU3COJDxynn"
      },
      "source": [
        "# Task 2: Mouse phenome database\n",
        "\n",
        "For this task, you will download and import some files from the Mouse Phenome Database (MPD, https://phenome.jax.org). This is a database containing a large amount of mouse experimental data, where information on several phenotypes/genotypes can be found.\n",
        "\n",
        "Execute the wget commands to download the following files to the colab session and the pd.read_csv commands to load them into python.\n",
        "\n",
        "Here is a short description of the file contents.\n",
        "\n",
        "- measurements.csv (= Description, units, and other metadata attributes for all phenotype measures (traits) in MPD. One row per measure.)\n",
        "- straininfo.csv (= Strain name, vendor, stock number, web page URL, MPD strain ID, and other attributes for each mouse strain in MPD. One row per strain.)\n",
        "- strainmeans.csv.gz (= Strain averages, SD, SEM, N, CV for all numeric strain survey phenotype measures in MPD.)\n",
        "- animaldatapoints.csv.gz (= Individual mouse readings for all strain survey phenotype measures. One row per animal reading.)\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "FYzpBoIrcopb"
      },
      "source": [
        "**Q2.1** How many different mouse strains are listed in the database? (1pt)\n",
        "\n",
        "**Q2.2** Several strains are not included in projects (n_proj is `nan`). How many strains were actually used in projects? (1pt)\n",
        "\n",
        "**Q2.3** Which strain was used in the most projects, give the name and id (1 pt)\n",
        "\n",
        "**Q2.4** Using the measurements and strainmeans file, look up the experiment with measnum = 29508. Give the description of the experiment and the average results for strain C3HeB/FeJ. (**hint:** You should get two values, one for males and one for females) (1 pts)\n",
        "\n",
        "**Q2.5** For the experiment above (29508) take the results from BALB/cByJ mice (strainid = 50) in animaldatapoints and perform a two-tailed two-sample t-test for `males > females`. What are the t-value and the p-value? (2 pts)\n",
        "\n",
        "**Q2.6** Plot a dotplot of the the body weight (measnum = 11201 listed in `measurements`) on the y-axis as a function of strain name on the x axis. Color-code sex. Add sensible title and axis labels. Make the axis legend lisible. (**hint:** strain ID is encoded differently in the tables!!!) (2 pts)\n",
        "\n",
        "**Q2.7** The project  \"Wahlsten1\" (projsym = \"Wahlsten1\" listed in `measurements`) examines anxiety behaviour using two assays (open field test) and elevated plus maze (epm) in 20 strains, and across both sexes. Plot the mean values across strain and sex for measnum =  10831 (percentage of time spent 10 cm from the wall, higher values indicate more anxiety) on the x-axis, and measnum = 10852 (percentage of time spent in open arms vs closed. lower values indicate less anxiety) on the y-axis. Add a regression line to the plot. Add sensible title and axis labels. Make the axis legend lisible. Get a correlation value for the two measurments. (3 pts)\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "98zGbhAgcopc"
      },
      "outputs": [],
      "source": [
        "# code to download the data\n",
        "\n",
        "! wget --no-check-certificate https://phenomedoc.jax.org/MPD_downloads/measurements.csv\n",
        "! wget --no-check-certificate https://phenomedoc.jax.org/MPD_downloads/straininfo.csv\n",
        "! wget --no-check-certificate https://phenomedoc.jax.org/MPD_downloads/strainmeans.csv.gz\n",
        "! wget --no-check-certificate https://phenomedoc.jax.org/MPD_downloads/animaldatapoints.csv.gz"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "FOkPUvqWcopc"
      },
      "outputs": [],
      "source": [
        "# code to load the data into pandas dataframes\n",
        "import pandas as pd\n",
        "measurements = pd.read_csv('measurements.csv')\n",
        "straininfo = pd.read_csv('straininfo.csv')\n",
        "strainmeans = pd.read_csv('strainmeans.csv.gz')\n",
        "animaldatapoints = pd.read_csv('animaldatapoints.csv.gz')"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "straininfo.head()"
      ],
      "metadata": {
        "id": "fN-NP7Re8lrQ"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "straininfo.head()\n",
        "total_strains = straininfo['strainname'].nunique()\n",
        "print(\"Q2.1: Total number of strains:\", total_strains)"
      ],
      "metadata": {
        "id": "2lphDum_68iz"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "strain_projects = straininfo.dropna(subset=['n_proj'])\n",
        "print(\"Q2.2: Number of strains used in projects:\", len(strain_projects))"
      ],
      "metadata": {
        "id": "MD9Bf6iT80Zn"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "most_strains = strain_projects.loc[strain_projects['n_proj'].idxmax()]\n",
        "print(most_strains)"
      ],
      "metadata": {
        "id": "BPJjP5Bf9RkS"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "Q2.3: The C57BL/6J strain has the most projects with 327."
      ],
      "metadata": {
        "id": "XIdoN4V4A6gI"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "measurement_info = measurements[measurements['measnum'] == 29508]\n",
        "print(measurement_info)\n"
      ],
      "metadata": {
        "id": "lyhzVl5yBvEA"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "results_C3HeB = strainmeans[(strainmeans['strain'] == 'C3HeB/FeJ') & (strainmeans['measnum'] == 29508)]\n",
        "print(results_C3HeB)"
      ],
      "metadata": {
        "id": "T3vvl4Q2Emjh"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "Q2.4: The research measures the percentage of cardiac ejection fractions with echocardiography in both male and female C3HeB/FeJ mice. The female mice have an average of 63.7 (SD: 7.48) and male 59.9 (SD: 6.45). So, the female mice have a higher percentage of cardiac ejections compared to male"
      ],
      "metadata": {
        "id": "HkMzRa2pD9TR"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "animaldatapoints.head()"
      ],
      "metadata": {
        "id": "TmeUa7toHnHu"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "results_BALB = animaldatapoints[(animaldatapoints['strainid'] == 50) & (animaldatapoints['measnum'] == 29508)]\n",
        "print(results_BALB)"
      ],
      "metadata": {
        "id": "3uAxnZLNIDeQ"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "from scipy.stats import ttest_ind\n",
        "female = results_BALB[results_BALB['sex'] == 'f']['measval']\n",
        "male = results_BALB[results_BALB['sex'] == 'm']['measval']\n",
        "\n",
        "\n",
        "t_stat, p_val = ttest_ind(male, female, equal_var=False)\n",
        "print(f\"T-statistiek: {t_stat}\")\n",
        "print(f\"P-waarde: {p_val}\")"
      ],
      "metadata": {
        "id": "wa-JHYXkI0p8"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "Q2.6 Plot a dotplot of the the body weight (measnum = 11201 listed in measurements) on the y-axis as a function of strain name on the x axis. Color-code sex. Add sensible title and axis labels. Make the axis legend lisible. (hint: strain ID is encoded differently in the tables!!!) (2 pts)"
      ],
      "metadata": {
        "id": "cilCs12rM8mn"
      }
    },
    {
      "cell_type": "code",
      "source": [],
      "metadata": {
        "id": "C0IojHvmM-Qu"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "measurements.head()"
      ],
      "metadata": {
        "id": "24afH1mjOhuB"
      },
      "execution_count": null,
      "outputs": []
    }
  ]
}